sdk/monitor/azure-monitor-opentelemetry-exporter/azure/monitor/opentelemetry/exporter/export/_base.py

```diff
@@ -205,19 +205,17 @@ def _transmit(self, envelopes: List[TelemetryItem]) -> ExportResult:
                 elif _is_redirect_code(response_error.status_code):
                     self._consecutive_redirects = self._consecutive_redirects + 1
                     if self._consecutive_redirects < self.client._config.redirect_policy.max_redirects:  # pylint: disable=W0212
-                        if response_error.response and response_error.response.headers:
-                            location = response_error.response.headers.get("location")
-                            if location:
-                                url = urlparse(location)
-                                if url.scheme and url.netloc:
-                                    # Change the host to the new redirected host
-                                    self.client._config.host = "{}://{}".format(url.scheme, url.netloc)  # pylint: disable=W0212
-                                    # Attempt to export again
-                                    result =  self._transmit(envelopes)
-                        if not self._is_stats_exporter():
-                            logger.error(
-                                "Error parsing redirect information."
-                            )
+                        location = response_error.response.headers.get("location")
```

Comment:
Can we still include the `if response_error.response and response_error.response.headers:` None checks?

Reply:
It's a bit tricky to avoid multiple nested if-statements. I could put it in the parent if block but that would skip the consecutive redirect counter check.
```python
elif _is_redirect_code(response_error.status_code) and response_error.response and response_error.response.headers:
```

Reply:
> It's a bit tricky to avoid multiple nested if-statements.

Don't worry about this too much. You can disable the linter if it complains.